com.unity.render-pipelines.core/Runtime/Volume/VolumeParameter.cs

```diff
@@ -1641,14 +1642,239 @@ internal override void Interp(VolumeParameter from, VolumeParameter to, float t)
+                    if (lhsValid && rhsValid)
+                    {
+                        lhsKey = lhsCurve.keys[lhsKeyCurr];
+                        rhsKey = rhsCurve.keys[rhsKeyCurr];
+
+                        if (lhsKey.time == rhsKey.time)
+                        {
+                            lhsKeyCurr++;
+                            rhsKeyCurr++;
+                        }
+                        else if (lhsKey.time < rhsKey.time)
+                        {
+                            // in this case:
+                            //     rhsKey[curr-1].time <= lhsKey.time <= rhsKey[curr].time
+                            // so interpolate rhsKey at the lhsKey.time.
+                            rhsKey = EvalKeyAtTime(rhsCurve.keys, rhsKeyCurr - 1, rhsKeyCurr, startTime, endTime, lhsKey.time);
+                            lhsKeyCurr++;
+                        }
+                        else if (lhsKey.time > rhsKey.time)
+                        {
+                            // this is the reverse of the lhs key case
+                            //     lhsKey[curr-1].time <= rhsKey.time <= lhsKey[curr].time
+                            // so interpolate lhsKey at the rhsKey.time.
+                            lhsKey = EvalKeyAtTime(lhsCurve.keys, lhsKeyCurr - 1, lhsKeyCurr, startTime, endTime, rhsKey.time);
+                            rhsKeyCurr++;
+                        }
+                        else
+                        {
+                            // should never happen, checking here to be safe
+                            Assert.IsTrue(false);
+                        }
+                    }
+                    else if (lhsValid)
+                    {
+                        // we are still processing lhsKeys, but we are out of rhsKeys, so increment lhs and evaluate rhs
+                        lhsKey = lhsCurve.keys[lhsKeyCurr];
+
+                        // rhs will be evaluated between the last rhs key and the extrapolated rhs key at the end time
+                        rhsKey = EvalKeyAtTime(rhsCurve.keys, rhsKeyCurr - 1, rhsKeyCurr, startTime, endTime, lhsKey.time);
+
+                        lhsKeyCurr++;
+                    }
+                    else if (rhsValid)
+                    {
+                        // we still have rhsKeys to lerp, but we are out of lhsKeys, to increment rhs and evaluate lhs
+                        rhsKey = rhsCurve.keys[rhsKeyCurr];
+
+                        // lhs will be evaluated between the last lhs key and the extrapolated lhs key at the end time
+                        lhsKey = EvalKeyAtTime(lhsCurve.keys, lhsKeyCurr - 1, lhsKeyCurr, startTime, endTime, rhsKey.time);
+
+                        rhsKeyCurr++;
+                    }
+                    else
+                    {
+                        // at least one of lhsValid or rhsValid should always be true
+                        Assert.IsTrue(false);
+                    }
```

Comment:
this also cannot happen
while (a && b) //of even (a || b)
{
   //you cannot have both a and b that are false here
}